### DISCUSSION TITLE
Looking for a way to set a chat users name on join.

### DISCUSSION THREAD

**Discussion by jonnyAnd** [owncast/owncast#3893] [Q&A]
I have owncast as part of a larger ecosystem and i load it in an iframe.
Often i know the user name of the user joining
I have been looking for a way that i can use the custom js functionality  (or any other way on the front end) to set the users displayed name.

any help appreciated, Thanks!

**Comment by gabek** [comment]
If you pass along the user display name in a HTTP header Owncast will use it when registering that user for the first time: https://owncast.online/docs/sslproxies/#inherit-display-name-from-reverse-proxy
